packages/xchain-util/src/types/asset.ts

```diff
@@ -4,4 +4,5 @@ export type Asset = {
   symbol: string
   ticker: string
   synth: boolean
+  trade: boolean
```

Comment:
im not sure if this is the correct way to go about adding a trade type to Asset type. @0xp3gasus, Thoughts? 
consider. 

A trade asset cannot be a synth Asset or vice versa. Currently only TC uses trade Assets.

Reply:
It is similar. see https://gitlab.com/thorchain/thornode/-/merge_requests/3326/diffs#e906cd37ce6a35a80301a5eeef02d48cea003cd7_120_123. 

There is an asset type and it can be either:
L1, Synth, Trade or derived.

Reply:
yes but it does not inherently support invalid state combinations 
A union type might work better here as with union types, inherently prevents invalid configurations through the type system, offering better safety and explicitness at the cost of increased complexity. 
example 
```
type BaseAsset = {
  chain: string;
  symbol: string;
  ticker: string;
};

type TradeAsset = BaseAsset & {
  synth: false;
  trade: true;
};

type SynthAsset = BaseAsset & {
  synth: true;
  trade: false;
};

type L1Asset = BaseAsset & {
  synth: false;
  trade: false;
};

type Asset = TradeAsset | SynthAsset | L1Asset;
```

Reply:
since it is a big change to the asset type,  I would like @0xp3gasus & @hippocampus-web3 to weigh in on this & options

Reply:
My concerns about the pull request. I think Stormzy approach is the one we have to follow but I would not stop just creating the `Asset` type as the union as the three different assets.
If we create this Asset type as the union, there are points in the library where we are going to work with an asset type that maybe is not support. I don't know much info about the aim that `TradeAssets` have so I don't know if this example is correct, but it can be used to understand what I mean. Following this nomenclature, nowadays, a swap can be done with `L1Assets` and `SynthAssets` using Thornode quote swap endpoint, will this endpoint work with `TradeAssets` too? If the answer is yes, this feature can work with the `Asset` type, but, if the answer is no, this feature can not work with the `Asset` type and has to work only with `L1Assets` and `SynthAssets`. Due to the big impact this change has, it's important to define what can and what can not be done with this new type of asset.

Reply:
agree with Oxp3gasus. Lets update thornode first and then see what features we can add to xchainjs with this trade account stuff.

Reply:
Thornode client has been updated to v1.128.1 in #1073

Reply:
[TradeAccounts](https://dev.thorchain.org/concepts/trade-accounts.html)

```Trade accounts provide professional traders (mostly arbitrage bots) a method to execute instant trades on THORChain without involving Layer1 transactions on external blockchains. Trade Accounts creates a new type of asset, backed by the network security rather than the liquidity in a pool (Synthetics), or by the RUNE asset (Derived Assets).```

* also trade acc are not activated yet 

Since this asset type is not for the average user or interface we should ask ourselves is it worth providing access too.

Reply:
> My concerns about the pull request. I think Stormzy approach is the one we have to follow but I would not stop just creating the `Asset` type as the union as the three different assets. If we create this Asset type as the union, there are points in the library where we are going to work with an asset type that maybe is not support. I don't know much info about the aim that `TradeAssets` have so I don't know if this example is correct, but it can be used to understand what I mean. Following this nomenclature, nowadays, a swap can be done with `L1Assets` and `SynthAssets` using Thornode quote swap endpoint, will this endpoint work with `TradeAssets` too? If the answer is yes, this feature can work with the `Asset` type, but, if the answer is no, this feature can not work with the `Asset` type and has to work only with `L1Assets` and `SynthAssets`. Due to the big impact this change has, it's important to define what can and what can not be done with this new type of asset.

Yes, you should be able to do swaps with trade assets, limited to TA<->TA and TA<->RUNE. 
A swap to and from a L1, from a TA is not allowed. They are a deposit or withdrawal.